recipes-openxt/xenclient/xenclient-uivm-xsessionconfig/ui-functions

```diff
@@ -0,0 +1,58 @@
+#! /bin/bash
+
+# Usage: xenmgr_get_string arg
+#   Print the requested Xenmgr UI configuration (arg) on stdout.
+xenmgr_config_ui_get_string() {
+    local arg="$1"
+    local path="/"
+    local dest="com.citrix.xenclient.xenmgr"
+    local intf="org.freedesktop.DBus.Properties.Get"
+
+    DBUS_SYSTEM_BUS_ADDRESS=tcp:host=1.0.0.0,port=5555 \
+    LD_PRELOAD=/usr/lib/libv4v-1.0.so.0 \
+    INET_IS_V4V=1 \
+        dbus-send --system --print-reply --dest="${dest}" "${path}" "${intf}" \
+            "string:com.citrix.xenclient.xenmgr.config.ui" \
+            "string:${arg}" \
+        | sed -ne 's|.*string "\(.\+\)".*|\1|p'
+}
+
+# Usage: xenmgr_get_locale
+#   Print the locale configured for Xenmgr on stdout.
+xenmgr_get_locale() {
+    local xenmgr_lang="$(xenmgr_config_ui_get_string language)"
```

Comment:
Should we provide a fallback if the above call fails to avoid echoing "_.UTF-8"?  Or would things have to be so hosed up, it is not worth bothering?